axelor-account/src/main/java/com/axelor/apps/account/service/moveline/MoveLineToolServiceImpl.java

```diff
@@ -358,4 +365,48 @@ public void checkDateInPeriod(Move move, MoveLine moveLine) throws AxelorExcepti
       }
     }
   }
+
+  @Transactional
+  public void setAmountRemainingReconciliableMoveLines(String startDate, Long accountId) {
+    List<Map> idMoveLinesList;
+    int count = 0;
+
+    if (startDate == null) {
+      idMoveLinesList =
+          this.moveLineRepository
+              .all()
+              .filter("self.accountId = :accountId")
+              .bind("accountId", accountId)
+              .order("id")
+              .select("id")
+              .fetch(0, 0);
```

Comment:
Don't use a fetch and don't iterate on a list, if you fetch a million moves, it will be so slow. 
Use an update, per example:
this.moveLineRepository
              .all()
              .filter("self.accountId = :accountId")
              .bind("accountId", accountId)
              .update("amountRemaining", result);

Reply:
We need to fetch moves to get credit  and debit to use it after in treatment